src/hooks/useDisclosure/useDisclosure.ts

```diff
@@ -12,24 +12,24 @@ export const useDisclosure = (
 ): UseDisclosureReturn => {
   const [isOpen, setIsOpen] = useState(initialState)
 
-  const open = (): void => {
+  const open = useCallback(() => {
     if (!isOpen) {
       setIsOpen(true)
       callbacks?.onOpen?.()
     }
```

Comment:
This would let you remove the dependency on isOpen for these callbacks, which would result in them never being redefined.
```suggestion
  const open = useCallback(() => {
    setIsOpen((_isOpen) => {
      if (!_isOpen) callbacks?.onOpen?.()
      return true
    })
```

Reply:
It's okay to do a setIsOpen to `true` on every invocation since true === true, which will not trigger any rerender or new state.

Reply:
Nice one!